### PR TITLE
download/save: save downloaded urls as files

### PR DESCRIPTION
**Pull request by ahmetalpbalkan**

### CODE REVIEW

File: download/save.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package download
+
+import "os"
+import "github.com/pkg/errors"
+import "io"
```
Comment (boumenot) [inline download/save.go]: Fix order.

File: download/save.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+const (
+	writeBufSize = 1024 * 2014
```
Comment (boumenot) [inline download/save.go]: I would have thought 8k or 16k, but that's an odd choice.

File: download/save_test.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	require.Nil(t, err)
+
+	fi, err := os.Stat(path)
+	require.Nil(t, err)
+	require.EqualValues(t, 128, fi.Size())
+	require.Equal(t, os.FileMode(0600), fi.Mode().Perm(), "mode should not be changed")
+}
+
+func TestSave_largeFile(t *testing.T) {
+	srv := httptest.NewServer(httpbin.GetMux())
```
Comment (boumenot) [inline download/save_test.go]: How long does it take to run this test?

**Comment by boumenot** [comment]
Few questions, but LGTM.

File: download/save.go (hunk 2 above)
Comment (ahmetalpbalkan) [inline download/save.go]: Hmm, I could use advice. I was planning to use io.CopyBuffer and let go decide the buffer size. It appears like the old code has 1 KB of buffer size.

File: download/save_test.go (hunk 3 above)
Comment (ahmetalpbalkan) [inline download/save_test.go]: Roughly 2.7-3s regardless of buffer size being 1k or 1mb. When I save to /dev/null, it's about 0.5s. So I think it is  pretty much the disk write speed on my laptop?

File: download/save.go (hunk 2 above)
Comment (boumenot) [inline download/save.go]: My vote is 8/16k.

Comment (ahmetalpbalkan) [inline download/save.go]: done.

File: download/save_test.go (hunk 3 above)
Comment (ahmetalpbalkan) [inline download/save_test.go]: travis build is faster:

https://travis-ci.org/Azure/custom-script-extension-linux/builds/145643847

=== RUN   TestSave_largeFile
--- PASS: TestSave_largeFile (0.39s)

**Pull request merged**